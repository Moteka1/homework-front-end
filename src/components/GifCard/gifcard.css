.gifcard-container {
  margin: 8px;
  cursor: pointer;
  transition: transform 0.2s;
}

.gifcard-container:hover {
  transform: scale(1.2);
  box-shadow: 3px 3px 5px 6px rgba(0, 0, 0, 0.4);
}

.modal-title {
  color: blue;
}

.modal-img {
  max-width: 100%;
  height: auto;
}
