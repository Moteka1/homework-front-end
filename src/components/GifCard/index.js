import React, { Component } from 'react';
import Modal from 'react-responsive-modal';
import './gifcard.css';

// card should include image info should be passed in as props from parent

export default class GifCard extends Component {
  // open and close modal
  state = {
    open: false
  };
  // functions to open and close modal
  openModal = () => {
    this.setState({ open: true });
  };

  onCloseModal = e => {
    e.stopPropagation();
    this.setState({ open: false });
  };

  render() {
    const { gifSrc, gifAlt, title, username, rating, giphyUrl } = this.props;
    const { open } = this.state;
    return (
      // need a way to open and close modal
      <div className="gifcard-container" onClick={this.openModal}>
        <img src={gifSrc} alt={gifAlt} />
        {/* conditionally rendering components */}
        <Modal open={open} onClose={this.onCloseModal} center>
          <h2 className="modal-title">{title}</h2>
          <img src={gifSrc} alt={gifAlt} />
          <div>
            <strong>Username:</strong> {username}
            <br />
            <strong>Rating:</strong> {rating}
            <br />
            <strong>Giphy Link:</strong>{' '}
            <a href={`${giphyUrl}`}>Link to the actual giphy!</a>
          </div>
        </Modal>
      </div>
    );
  }
}
