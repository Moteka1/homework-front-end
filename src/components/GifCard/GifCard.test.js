import React from 'react';
import GifCard from './index.js';
import renderer from 'react-test-renderer';

describe('<GifCard />', () => {
  it('matches the snapshot', () => {
    var tree = renderer.create(<GifCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
