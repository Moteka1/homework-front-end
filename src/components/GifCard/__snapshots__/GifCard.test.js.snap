// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCard /> matches the snapshot 1`] = `
<div
  className="gifcard-container"
  onClick={[Function]}
>
  <img />
</div>
`;
